### PR TITLE
enh(secu): sanitize and cast parameters to int

### PR DESCRIPTION
**Pull request by Sjord**
Convert the num, limit, and period parameters to int. Sanitize them and call
intval on them, when performing the XML export on the event logs page.

These parameters are certainly meant to be numeric, since we do calculations
with them later on, such as this:

    $limitReq = " LIMIT " . $num * $limit . ", " . $limit;

Casting them to int prevents the possibility for SQL injection in the line
above. Additionally, on line 656 we loop over values derived from $num, and
that can lead to an infinite loop if $num is not numeric.

This change causes no functional changes to the event logs XML export. When
calling this page with non-numeric values for num, limit, or period, the
default options are used instead.

fixe (MON-5477)

### CODE REVIEW

File: www/include/eventLogs/xml/data.php [hunk 1]
```diff
@@ -170,13 +170,13 @@
     );
 }
 
-$num = isset($inputs["num"]) ? htmlentities($inputs["num"]) : "0";
-$limit = isset($inputs["limit"]) ? htmlentities($inputs["limit"]) : "30";
+$num = isset($inputs["num"]) ? intval($inputs["num"]) : "0";
```
Comment (callapa) [inline www/include/eventLogs/xml/data.php:173]:
```suggestion
$num = isset($inputs["num"]) ? (int) $inputs["num"] : 0;
```

File: www/include/eventLogs/xml/data.php [hunk 2]
```diff
@@ -170,13 +170,13 @@
...
+$limit = isset($inputs["limit"]) ? intval($inputs["limit"]) : "30";
```
Comment (callapa) [inline www/include/eventLogs/xml/data.php:174]:
```suggestion
$limit = isset($inputs["limit"]) ? (int) $inputs["limit"] : 30;
```

File: www/include/eventLogs/xml/data.php [hunk 3]
```diff
@@ -170,13 +170,13 @@
...
 $StartDate = isset($inputs["StartDate"]) ? htmlentities($inputs["StartDate"]) : "";
 $EndDate = isset($inputs["EndDate"]) ? $EndDate = htmlentities($inputs["EndDate"]) : "";
 $StartTime = isset($inputs["StartTime"]) ? $StartTime = htmlentities($inputs["StartTime"]) : "";
 $EndTime = isset($inputs["EndTime"]) ? $EndTime = htmlentities($inputs["EndTime"]) : "";
-$auto_period = isset($inputs["period"]) ? $auto_period = htmlentities($inputs["period"]) : "-1";
+$auto_period = isset($inputs["period"]) ? $auto_period = intval($inputs["period"]) : "-1";
```
Comment (callapa) [inline www/include/eventLogs/xml/data.php:179]:
```suggestion
$auto_period = isset($inputs["period"]) ? (int) $inputs["period"] : -1;
```

**Comment by Sjord** [comment]
Thanks, @callapa. Why do you prefer `(int)` over `intval`?

**Comment by Sjord** [comment]
I have added a commit with your suggested changes.

**Comment by callapa** [comment]
>Thanks, @callapa. Why do you prefer (int) over intval?

(int) is faster than intval() and as we use base 10, we prefer to use (int)

**Pull request merged**